### PR TITLE
docs: add more information about metrics

### LINKED ISSUE #2042: Karpenter metrics still not detailed
> The metrics are now documented on karpenter.sh, you can see the preview docs at https://karpenter.sh/preview/tasks/metrics/ and they'll be on the main site after the next release.

All good and fine, but what is the endpoint? On what port do we access it? can we filter for each metric that is there? Also can i scrape the metrics with Datadog? Is it necessary with Prometheus?

_Originally posted by @cristiangstefan in https://github.com/aws/karpenter/issues/1275#issuecomment-1174421488_

### PR DESCRIPTION
**Pull request by tzneal**
Fixes #2042

**Description**

Add more information about the metrics endpoint and document the default
values for the command line parameters.

**How was this change tested?**

Website preview

**Does this change impact docs?**
- [X] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [ ] No

**Release Note**

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: hack/docs/metrics_gen_docs.go [hunk 1]
```diff
@@ -114,7 +114,8 @@ description: >
 ---
 `)
 	fmt.Fprintf(f, "<!-- this document is generated from hack/docs/metrics_gen_docs.go -->\n")
-	fmt.Fprintf(f, "Karpenter writes several metrics to Prometheus to allow monitoring cluster provisioning status\n")
+	fmt.Fprintf(f, "Karpenter makes several metrics available in Prometheus format to allow monitoring cluster provisioning status. "+
+		"These metrics are available on the metrics port at `/metrics` configurable via the `METRICS_PORT` environment variable documented [here](../configuration)\n")
```
Comment (ellistarn) [inline hack/docs/metrics_gen_docs.go:118]:
```suggestion
		"These metrics are available on the metrics port at `karpenter-metrics.karpenter.cluster.local:8081/metrics` configurable via the `METRICS_PORT` environment variable documented [here](../configuration)\n")
```

Comment (tzneal) [reply hack/docs/metrics_gen_docs.go]: We don't expose a karpenter-metrics service.

**Review by bwagner5 — APPROVED** [review]
Nice!

File: hack/docs/metrics_gen_docs.go (hunk 1 above)
Comment (ellistarn) [reply hack/docs/metrics_gen_docs.go]: Oh -- we used to. SGTM.

**Pull request merged**